### DISCUSSION TITLE
How to trigger workflow run in GitHub Actions

### DISCUSSION THREAD

**Discussion by drey0143143** [community/community#37553] [Actions] [answered]
I'm able to trigger plan1 workflow when a pull request has label name "bug" but this also triggers plan2 job at the same time. I only want plan2 to run when there is a pull request with no label. please how can I achieve this?
```
name: Testing versions upgrades 
on:
  pull_request: 
    types: [ labeled, unlabeled ]

jobs:
  plan1:
    name: "bug_fix"
    if: github.event.label.name == 'bug'

 plan2:
   name: deploy
   if:github.event.action == unlabeled

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @drey0143143 ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. I've gone ahead and moved it for you. Good luck!

**Reply by drey0143143** [reply]
@queenofcorgis  thanks

**Comment by antonblr** [comment]
Hey @drey0143143 - is it not working as it is right now? The `if: github.event.action == "unlabeled"` condition should make `plan2` job to run only if the label was removed. 

Also, you can use this if you want to check for any labels for PR - `${{ join(github.event.pull_request.labels.*.name, '') }}`. It will be empty if there is no labels.
To check specific label - `if: contains(github.event.pull_request.labels.*.name, 'bug')`

**Comment by drey0143143** [comment]
@antonblr I just tried this if: contains(github.event.pull_request.labels.*.name, 'bug') and it worked as expected but plan2 is also running at the same time with "${{ join(github.event.pull_request.labels.*.name, '') }}" or "if: github.event.action == unlabeled"  which is not what i expected. I only want plan2 to run when a pull request is created without any labels.

**Comment by antonblr** [comment, marked as answer]
@drey0143143 - simply negate the first condition:

`if: ! contains(github.event.pull_request.labels.*.name, 'bug')` (note the `!`)

Or, if you want to run `plan2` when there is no labels at all:

`if: join(github.event.pull_request.labels.*.name, '') == ''`  (you need to compare the result of `join()` with something to make it a condition)

**Comment by drey0143143** [comment]
@antonblr  if: join(github.event.pull_request.labels.*.name, '') == '' solve the issue. thank you so much...... I have one last issue I'm now facing

I want to be able to run push_master_workflow1 only when the PR is labeled "bug" and run push_master_workflow2 when there no labels at all on the PR.

**Reply by antonblr** [reply]
I'm not sure there is a way of doing so at the workflow level - it gets triggered by `on:` directive and there is no place for conditions. 

I think the bet for now is to have a single workflow with multiple jobs and `if:` conditions for every job.

**Comment by drey0143143** [comment]
@antonblr this is what I currently have as per your recommendation but both did not work
```
name: push to master
on:
  push:
   branches:
    - master

jobs:
  push1:
    name: "run if PR merging has a "bug" label
    if: github.event_name == "push" && github.event.pull_request.labels.*.name == 'bug'
  push2:
    name: "run if PR merging has no label at all"
    if: github.event_name == "push" && join(github.event.pull_request.labels.*.name, '') == ''

**Reply by antonblr** [reply]
I didn't recommend mixing `push` events with `pull_request`. Because `push` event is not `pull_request` event. In other words, `github.event.pull_request` data might not be available at all. Definitely this won't work.

**Comment by drey0143143** [comment]
@antonblr  that is right you didn't recommend such.   my question was suppose to be, how can i trigger both jobs based on if one pull request that is merging has a label and trigger the other when the pull request merging doesnt have a label at all. Thanks

**Comment by antonblr** [comment]
>how can i trigger both jobs based on if one pull request that is merging has a label and trigger the other when the pull request merging doesnt have a label at all

Perhaps you can do something like that (trigger for closed PRs and then check if it was `merged == true` and had label(s)):
```yaml
name: Pushed to master
on:
  pull_request:
    types:
      - closed

jobs:
  is-labeled:
    if: github.event.pull_request.merged == true && contains(github.event.pull_request.labels.*.name, 'bug')
    ...
 
  not-labeled:
    if: github.event.pull_request.merged == true && ! contains(github.event.pull_request.labels.*.name, 'bug') # or replace with join() == ''
    ...

**Reply by drey0143143** [reply]
@antonblr thanks your answer worked.
